src/Symfony/Component/Cache/Adapter/AbstractMemcacheAdapter.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Rob Frawley 2nd <rmf@src.run>
+ */
+abstract class AbstractMemcacheAdapter extends AbstractAdapter
```

Comment:
this should be turned into a trait with the `@internal` annotation (see other traits in the Adapter folder)